tests/0113-cooperative_rebalance.cpp

```diff
@@ -2994,6 +3029,78 @@ static void r_lost_partitions_commit_illegal_generation_test_local() {
   test_mock_cluster_destroy(mcluster);
 }
 
+/* Check commit is retried on FENCED_MEMBER_EPOCH, using new epoch taken
+ * from HB. */
+static void t_consumer_group_consumer_retry_commit_on_fenced_member_epoch() {
+  const char *bootstraps;
+  rd_kafka_mock_cluster_t *mcluster;
+  const char *groupid = "mygroup";
+  const char *topic   = "test";
+  const int msgcnt    = 100;
+  rd_kafka_t *c;
+  rd_kafka_conf_t *conf;
+  rd_kafka_topic_partition_list_t *rktpars =
+      rd_kafka_topic_partition_list_new(1);
+
+  SUB_TEST();
+
+  mcluster = test_mock_cluster_new(3, &bootstraps);
+
+  rd_kafka_mock_coordinator_set(mcluster, "group", groupid, 1);
+
+  /* Seed the topic with messages */
+  test_produce_msgs_easy_v(topic, 0, 0, 0, msgcnt, 10, "bootstrap.servers",
+                           bootstraps, "batch.num.messages", "10",
+                           "security.protocol", "plaintext", NULL);
+
+  test_conf_init(&conf, NULL, 30);
+  test_conf_set(conf, "bootstrap.servers", bootstraps);
+  test_conf_set(conf, "security.protocol", "PLAINTEXT");
+  test_conf_set(conf, "group.id", groupid);
+  test_conf_set(conf, "auto.offset.reset", "earliest");
+  test_conf_set(conf, "enable.auto.commit", "false");
+  test_conf_set(conf, "partition.assignment.strategy", "cooperative-sticky");
+
+  c = test_create_consumer(groupid, rebalance_cb, conf, NULL);
+
+  test_consumer_subscribe(c, topic);
+
+  expect_rebalance("initial assignment", c,
+                   RD_KAFKA_RESP_ERR__ASSIGN_PARTITIONS,
+                   rd_false /*don't expect lost*/, 5 + 2);
+
+
+  /* Consume some messages so that the commit has something to commit. */
+  test_consumer_poll("consume", c, -1, -1, -1, msgcnt / 2, NULL);
+
+  /* Fail Commit */
+  rd_kafka_mock_push_request_errors(mcluster, RD_KAFKAP_OffsetCommit, 5,
+                                    RD_KAFKA_RESP_ERR_FENCED_MEMBER_EPOCH,
+                                    RD_KAFKA_RESP_ERR_FENCED_MEMBER_EPOCH,
+                                    RD_KAFKA_RESP_ERR_FENCED_MEMBER_EPOCH,
+                                    RD_KAFKA_RESP_ERR_FENCED_MEMBER_EPOCH,
+                                    RD_KAFKA_RESP_ERR_FENCED_MEMBER_EPOCH);
+
+  rd_kafka_commit(c, NULL, rd_false);
+
+  TEST_CALL_ERR__(rd_kafka_committed(c, rktpars, 2000));
```

Comment:
This might fail intermittently as we are not waiting for the commit to pass before fetching the offsets.

Reply:
But previous commit is sync, why we are not waiting for the commit to pass?

Reply:
My bad I think. I read the last parameter as `sync` instead.